### ISSUE TITLE
Send a push and wait for response

### ISSUE THREAD

**Issue by seaders** [randomchars/pushbullet.py#49]
So I've just set up a bot that does stuff 24/7, and it currently automates some things, and notifies me when it does via this wonderful library via the wonderful PushBullet.  *But*, there's some things that I'm not sure I know what I want it to do until I see what it has.

Think of a tweet.  Tweet from ASpecialTweeter comes in, and I always want to retweet that.  But if a tweet comes in from ANotSoSpecialTweeter, I only want to retweet after I've seen the tweet.  So, in a perfect World, what I'd love is what you get in the Firefox plugin, but also on mobile, from one of my own tweets.

Basically, if it's at all possible, I'd love to, via the Python library, do something like,

    resp = pb.push_interactive_note('Hey seaders, ANotSoSpecialTweeter just tweeted "{}", whatcha wanna do about it?'.format(tweet.text),
                                    'Retweet', 'Favourite', onDismiss='Dismiss')
    if 'Retweet' == resp:
        twitter.retweet(tweet.id)
    ....

In my head, the power this would give this side of things would be amazing.

I run more than a few services like this, and being able to have such an easy way to program different things on servers, then ask myself what I want to do, via PB would add an extra dimension of power again.

**Comment by r-darwish** [comment]
The question is which push should be treated as response to the initial push. If we treat the next push, whatever its content is, as a response to your push then another push coming from some other bot (or even this bot) can be treated by mistake as a response.

A possible idea is that push_interactive_note will generate some random ID and stick it to the message, then, only when it receives a push containing this ID it will treat it as the response. So, for example, you'll receive something like:

#325: Do you want to retweet 'x'?

Then you'll have to answer:
#325: yes

I'm open to further suggestions.

**Comment by seaders** [comment]
Yep, for an initial version, that would be great and more than acceptable.

Ideally in the long run, if you architect the system that way, you'd have something like direct response to a push, you know, not something on the main thread, but responses that are more like child responses to the main push.
